client/client.go

```diff
@@ -180,17 +201,18 @@ func (client *Client) readLoop() {
 
 		contents := make([]byte, length)
 
-		if _, err = io.ReadFull(client.rw, contents); err != nil {
+		if _, err = io.ReadFull(client.conn, contents); err != nil {
 			if err = client.reconnect(err); err != nil {
 				break
 			}
 
 			continue
 		}
 
-		resp = getResponse()
+		resp = client.responsePool.Get().(*Response)
```

Comment:
This seems to rely on client.process() always calling client.responsePool.Put() at the end, which it seems to do but could be brittle. Could defer do the trick?

Reply:
The pool doesn't have a cap, I take it?